tests/integration/test_authd/test_authd_force_options.py

```diff
@@ -0,0 +1,252 @@
+    try:
+        SSL_socket.open()
+    except ssl.SSLError as exception:
+        # We did not expect this error, fail test
+        raise
+    SSL_socket.send(message, size=False)
+    response = SSL_socket.receive().decode()
+    SSL_socket.close()
+    return response
+
+
+# Initial clean client_keys file
+# Stop Wazuh
+control_service('stop')
+
+# reset_client_keys
+clean_client_keys_file()
+
+# Start Wazuh
+control_service('start')
+
+"""Wait until agentd has begun"""
+
+
+def callback_agentd_startup(line):
+    if 'Accepting connections on port 1515' in line:
+        return line
+    return None
+
+
+log_monitor = FileMonitor(LOG_FILE_PATH)
+log_monitor.start(timeout=30, callback=callback_agentd_startup)
+
+
+# @pytest.mark.parametrize('test_case', [case['test_case'] for case in ssl_configuration_tests])
+def test_authd_force_options(get_configuration, configure_environment, configure_sockets_environment):
+    """Check that every input message in authd port generates the adequate output
+
+    Parameters
+    ----------
+    test_case : list
+        List of test_cases, dict with following keys:
+            - input: message that will be tried to send to the manager
+            - output: expected response
+            - insert_prev_agent: yes or no (for duplicated ip or name cases)
+                1) if insert_prev_agent_custom is present: previous input message is overwrite by the custom message
+                    (insert_prev_agent_custom: "OSSEC A:'user0' IP:'10.10.10.10'")
+                2) if insert_prev_agent_custom is not present: send the masage equals to input
+            - insert_random_pass_in_query:
+              "yes" if is needed add random pass to input query (for register with random pass cases)
+            - insert_hostname_in_query:
+              "yes" if is present add host name to input message
+    """
+    current_test = get_current_test()
+    test_case = force_options_tests[current_test]['test_case']
+
+    override_wazuh_conf(get_configuration)
+    for config in test_case:
+
+        # insert previous agent to force repeated case
+        try:
+            if config['insert_prev_agent'] == "yes":
+                try:
+                    response = send_message(config['insert_prev_agent_custom'])
+                except KeyError:
+                    response = send_message(config['input'])
+
+                # Prev output is expected
+                expected = "OSSEC K:'"
+                assert response, \
+                    'Failed connection previous insert for {}: {}'.format \
+                        (force_options_tests[current_test]['name'], config['input'])
+                assert response[:len(expected)] == expected, \
+                    "Failed response previous '{}': Input: {}".format \
+                        (force_options_tests[current_test]['name'], config['input'])
+                if expected == "OSSEC K:'":
+                    time.sleep(0.5)
```

Comment:
Hardcoded timeout is not allowed. Please define it as a variable at the beginning of the module. If this will be used for other tests, please defined it in a common library.

Reply:
Fixed.